core/src/test/scala/unit/kafka/server/OffsetDeleteRequestTest.scala

```diff
@@ -0,0 +1,182 @@
+    }
+
+    // Creates the __consumer_offsets topics because it won't be created automatically
+    // in this test because it does not use FindCoordinator API.
+    createOffsetsTopic()
+
+    // Create the topic.
+    createTopic(
+      topic = "foo",
+      numPartitions = 3
+    )
+
+    for (version <- ApiKeys.OFFSET_DELETE.oldestVersion() to ApiKeys.OFFSET_DELETE.latestVersion(isUnstableApiEnabled)) {
+      // Join the consumer group. Note that we don't heartbeat here so we must use
+      // a session long enough for the duration of the test.
+      val (memberId, memberEpoch) = joinConsumerGroup(
+        groupId = "grp",
+        useNewProtocol = useNewProtocol
+      )
+
+      // Commit offsets.
+      for (partitionId <- 0 to 2) {
+        commitOffset(
+          groupId = "grp",
+          memberId = memberId,
+          memberEpoch = memberEpoch,
+          topic = "foo",
+          partition = partitionId,
+          offset = 100L + partitionId,
+          expectedError = Errors.NONE,
+          version = ApiKeys.OFFSET_COMMIT.latestVersion(isUnstableApiEnabled)
+        )
+      }
+
+      // Delete offset with topic that the group is subscribed to.
+      deleteOffset(
+        groupId = "grp",
+        topic = "foo",
+        partition = 0,
+        expectedPartitionError = Errors.GROUP_SUBSCRIBED_TO_TOPIC,
+        version = version.toShort
+      )
+
+      // Unsubscribe the topic.
+      if (useNewProtocol) {
+        consumerGroupHeartbeat(
+          groupId = "grp",
+          memberId = memberId,
+          memberEpoch = memberEpoch,
+          subscribedTopicNames = List.empty
+        )
+      } else {
+        leaveGroup(
+          groupId = "grp",
+          memberId = memberId,
+          useNewProtocol = false
+        )
+      }
+
+      // Delete offsets.
+      for (partitionId <- 0 to 1) {
+        deleteOffset(
+          groupId = "grp",
+          topic = "foo",
+          partition = partitionId,
+          version = version.toShort
+        )
+      }
+
+      // Delete offsets with partition that doesn't exist.
+      deleteOffset(
+        groupId = "grp",
+        topic = "foo",
+        partition = 5,
+        expectedPartitionError = Errors.UNKNOWN_TOPIC_OR_PARTITION,
+        version = version.toShort
+      )
+
+      // Delete offset with unknown group id.
+      deleteOffset(
+        groupId = "grp-unknown",
+        topic = "foo",
+        partition = 2,
+        expectedResponseError = Errors.GROUP_ID_NOT_FOUND,
+        version = version.toShort
+      )
+
+      // Delete offset with empty group id.
+      deleteOffset(
+        groupId = "",
+        topic = "foo",
+        partition = 2,
+        expectedResponseError = Errors.INVALID_GROUP_ID,
+        version = version.toShort
+      )
+
+      // Delete offset with both invalid id and invalid topic should return with a response with top-level error.
+      deleteOffset(
```

Comment:
Could we also add similar tests for the offset fetch as we had a similar issue there?